src/ucc/api/ucc.h

```diff
@@ -1574,5 +1588,226 @@ static inline ucc_status_t ucc_collective_test(ucc_coll_req_h request)
+ * events. All events must be acknowledged using the @ref ucc_ee_ack_event
+ * interface. The event acknowledged is destroyed by the library. An event
+ * fetched with @ref ucc_ee_get_event but not acknowledged might consume
+ * resources in the library.
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_get_event(ucc_ee_h ee, ucc_ev_t *ev);
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine acks the events from the event queue.
+ *
+ * @param [in]  ee      execution engine handle
+ * @param [in]  ev      Event to be acked
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * An event acknowledged by the user using @ref ucc_ee_ack_event is destroyed by
+ * the library. Any triggered operations on the event should be completed before
+ * calling this interface. The behavior is undefined if the user acknowledges
+ * the event while waiting on the event or triggering operations on the event.
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_ack_event(ucc_ee_h ee, ucc_ev_t ev);
+
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine to set the event to the tail of the queue.
+ *
+ * @param [in]  ee        execution engine handle
+ * @param [in]  ev        Event structure fetched from the event queue
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * @ref ucc_ee_set_event sets the event on the execution engine. If the
+ * operations are waiting on the event when the user sets the event, the
+ * operations are launched. The events created by the user need to be destroyed
+ * by the user.
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_set_event(ucc_ee_h ee, ucc_ev_t ev);
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine blocks the calling thread until there is an event on the queue.
+ *
+ * @param [in]  ee        execution engine handle
+ * @param [out] ev        Event structure fetched from the event queue
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * The user thread invoking the @ref ucc_ee_wait interface is blocked until an
+ * event is posted to the execution engine.
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_wait(ucc_ee_h ee, ucc_ev_t *ev);
+
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine posts the collective operation on the execution engine, which is
+ * launched on the event.
+ *
+ * @param [in]  ee          execution engine handle
+ * @param [in]  ee_event    Event triggering the post operation
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * @ref ucc_collective_triggered_post allow the users to schedule a collective
+ * operation that executes in the future when an event occurs on the execution
+ * engine.
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_collective_triggered_post(ucc_ee_h ee, ucc_ev_t ee_event);
```

Comment:
@manjugv @bureddy what is the plan for triggering operations on GPU streams? cuStreamWriteVal and cuStreamWaitVal or through kernels that wait for a value to be set?

Reply:
Addressed in the WG discussion.